agent/connect/ca/provider.go

```diff
@@ -2,10 +2,21 @@ package ca
 
 import (
 	"crypto/x509"
+	"log"
+	"time"
 )
 
 //go:generate mockery -name Provider -inpkg
 
+// NeedsLogger is an interface that allows Consul to pass a configured
+// logger into any component that needs one.
+type NeedsLogger interface {
+	// SetLogger tells the provider to use the specified logger.
+	// Note: this should be one of the first calls made after instantiating
```

Comment:
```suggestion
	// This is called immediately after instantiating
```

Reply:
We're in control of when it's called so we just need to tell implementers in the future what they can rely on.